File: AdminPanelUI/src/types/eventPage.ts
export const EVENT_PAGE_FIELD_TYPES = [
  { value: 'Text', label: 'نص' },
  { value: 'Number', label: 'رقم' },
  { value: 'SingleSelect', label: 'اختيار واحد' },
  { value: 'MultiSelect', label: 'اختيار متعدد' },
] as const

export type EventPageFieldType = (typeof EVENT_PAGE_FIELD_TYPES)[number]['value']

export interface EventPageTrack {
  id: number
  title: string
  description: string | null
  sortOrder: number
}

export interface EventPageFormField {
  id: number
  label: string
  fieldType: EventPageFieldType
  isRequired: boolean
  sortOrder: number
  options: string[]
}

export interface EventPageListItem {
  id: number
  activityName: string
  slug: string
  courseTitle: string
  imageUrl: string | null
  isPublished: boolean
  isRegistrationOpen: boolean
  createdAt: string
}

export interface EventPageLookup {
  id: number
  activityName: string
}

export interface EventPage {
  id: number
  activityName: string
  slug: string
  courseTitle: string
  invitationText: string | null
  mosqueName: string | null
  subjectText: string | null
  dateText: string | null
  timeText: string | null
  extraNotes: string | null
  supervisorsText: string | null
  contactPhone: string | null
  socialAccounts: string | null
  locationNote: string | null
  imageUrl: string | null
  isPublished: boolean
  isRegistrationOpen: boolean
  createdAt: string
  tracks: EventPageTrack[]
  formFields: EventPageFormField[]
}

export interface SaveEventPageTrackPayload {
  title: string
  description?: string
  sortOrder: number
}

export interface SaveEventPageFieldPayload {
  id?: number
  label: string
  fieldType: EventPageFieldType
  isRequired: boolean
  sortOrder: number
  options: string[]
}

export interface SaveEventPagePayload {
  activityName: string
  slug: string
  courseTitle: string
  invitationText?: string
  mosqueName?: string
  subjectText?: string
  dateText?: string
  timeText?: string
  extraNotes?: string
  supervisorsText?: string
  contactPhone?: string
  socialAccounts?: string
  locationNote?: string
  isPublished: boolean
  isRegistrationOpen: boolean
  imageFile?: File
  tracks: SaveEventPageTrackPayload[]
  formFields: SaveEventPageFieldPayload[]
}

File: AdminAPI/Models/EventPage.cs
namespace AdminAPI.Models;

public class EventPage
{
    public int Id { get; set; }
    public string ActivityName { get; set; } = string.Empty;
    public string Slug { get; set; } = string.Empty;
    public string CourseTitle { get; set; } = string.Empty;
    public string? InvitationText { get; set; }
    public string? MosqueName { get; set; }
    public string? SubjectText { get; set; }
    public string? DateText { get; set; }
    public string? TimeText { get; set; }
    public string? ExtraNotes { get; set; }
    public string? SupervisorsText { get; set; }
    public string? ContactPhone { get; set; }
    public string? SocialAccounts { get; set; }
    public string? LocationNote { get; set; }
    public string? ImageUrl { get; set; }
    public bool IsPublished { get; set; } = true;
    public bool IsRegistrationOpen { get; set; } = true;
    public DateTime CreatedAt { get; set; }

    public ICollection<EventPageTrack> Tracks { get; set; } = new List<EventPageTrack>();
    public ICollection<EventPageFormField> FormFields { get; set; } = new List<EventPageFormField>();
    public ICollection<EventPageResponse> Responses { get; set; } = new List<EventPageResponse>();
}
